src/renderer/components/watch-video-playlist/watch-video-playlist.js

```diff
@@ -368,14 +363,10 @@ export default defineComponent({
         this.channelName = channelName
         this.channelId = playlist.info.author?.id
 
-        const videos = playlist.items.map(parseLocalPlaylistVideo)
-
-        while (playlist.has_continuation) {
-          playlist = await playlist.getContinuation()
-
-          const parsedVideos = playlist.items.map(parseLocalPlaylistVideo)
-          videos.push(...parsedVideos)
-        }
+        const videos = []
+        await untilEndOfLocalPlayList(playlist, (p) => {
```

Comment:
Using promises and callbacks together is a weird code pattern, which I'm not a fan of (the whole point of async/await in javascript was to get away from callbacks), but as we are in a rush and the code works, i'll accept it this time.